cross/fish/Makefile

```diff
@@ -1,20 +1,76 @@
 PKG_NAME = fish
-PKG_VERS = 3.7.1
+PKG_VERS = 4.0.2
 PKG_EXT = tar.xz
 PKG_DIST_NAME = $(PKG_NAME)-$(PKG_VERS).$(PKG_EXT)
 PKG_DIST_SITE = https://github.com/fish-shell/fish-shell/releases/download/$(PKG_VERS)
 PKG_DIR = $(PKG_NAME)-$(PKG_VERS)
 
-DEPENDS = cross/ncursesw cross/pcre2
+DEPENDS =
+
+# requirements: C++11 compiler (g++ 4.8 or later, or clang 3.3 or later)
+UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(OLD_PPC_ARCHS)
+# build with custom qoriq rust_target fails to build in spk folder (but not in cross/fish)
+UNSUPPORTED_ARCHS += $(PPC_ARCHS)
 
 HOMEPAGE = https://fishshell.com
 COMMENT = fish is a smart and user-friendly command line shell for Linux, macOS, and the rest of the family.
 LICENSE = GPL-2.0
 
-# requirements: C++11 compiler (g++ 4.8 or later, or clang 3.3 or later)
-UNSUPPORTED_ARCHS = $(ARMv5_ARCHS) $(OLD_PPC_ARCHS)
+POST_INSTALL_TARGET = fish_post_install
+
+ENV += FISH_BUILD_DOCS=0
+
+# for fish_test_helper.c
+ADDITIONAL_CFLAGS = -D_DEFAULT_SOURCE
+
+include ../../mk/spksrc.common.mk
+ifeq ($(call version_lt, $(TC_GCC), 5),1)
+ADDITIONAL_CFLAGS += -std=c11
+endif
+
+# fish fails to build with gettext
+CMAKE_ARGS = -DWITH_GETTEXT=OFF
+
+# fish >= 4.0.0 requires rust
+# IMPORTANT:
+# fish must still be built with cmake
+# direct builds with cargo/rust (using spksrc.cross-rust.mk) create
+# experimental static versions without data files
+
+# keep cross tools in ENV for cargo to find CC and LN of toolchain
+CMAKE_USE_TOOLCHAIN_FILE = OFF
```

Comment:
@hgy59 guessing you mean `LD`?
noting that I have additional changes to do to `CMAKE` to sync with meson and fortran's extra details. Maybe I could add an option to keep `CC` and `LD` defined when setting something such as `CMAKE_USE_RUST` or similar?

Reply:
yes, without this it was using `/usr/bin/ld`
`/usr/bin/gcc` could be overwritten with `ENV += TARGET_CC=$(TC_PATH)$(TC_PREFIX)gcc` but there is only `LD` for the linker.
This was required for pcre2/pcre2-sys and fish-printf.
IMO there is no need for special handling of _CMAKE_USE_RUST_ ...

What bothers me more is the different behaviour of the builds in cross/fish and spk/fish...